### PR TITLE
[Spot] Better spot logs

### PR DESCRIPTION
**Pull request by Michaelvll**
This is to simplify the spot logs when waiting for the logs to be ready.

Tested:
- [x] `sky spot launch --num-nodes=2 'sleep 10000'`
Current:
image

image

### CODE REVIEW

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:256]: How about "Waiting for the spot job to start .."? Without "to start" it reads a bit unclear.

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:252]:
```suggestion
                    status_help_str = f' (job status: {spot_status.value})'
```

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:203]:
```suggestion
    status_msg = ('[bold cyan]Waiting for controller process to start/to be RUNNING (status: {}).[/]'
```
Reasons:
- distinguish "controller" and "controller process" 
  - We may want to avoid this confusion by renaming "controller process" to "monitor process" (for future)
- "waiting for X" is unclear to me

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:213]: Q: why can we do `status_msg.update(status_str)`? `''.update()` says str doesn't have an update func?

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:205]: Q: I can't seem to trigger this log with `sky spot launch sleep 10000`.

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:218]: Q: I can't seem to trigger this log with `sky spot launch sleep 10000`.

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:288]: Q: I can't seem to trigger this log with sky spot launch sleep 10000.

Add "to start"?

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll - this helps the UX a bunch. Some questions.

(For the future: if more people start using spot_launch() , we may want to use the original logging?)

**Comment by concretevitamin** [comment]
Screen Shot 2022-11-18 at 18 16 42

I now see the `⠋ Waiting for the spot jobI 11-19...` line, which got printed during the log streaming of a `sleep 1000` job and after a `sky spot cancel -a` was issued.

The line joining can be fixed, but maybe we shouldn't show "Waiting for the spot job" in this case at all?

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: It is the general initial status to avoid empty logging and will be soon updated later. I changed it to `Waiting for controller process to be RUNNING`, so it is consistent with the first logging afterwards.

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: Done. Thanks!

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: Good point! Fxied. Thanks!

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: This only happens when the cluster is preempted, i.e. the `tail_logs` returns nonzero.

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: Done. Thanks!

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: Oops, that was a typo. Fixed. Thanks!

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py]: That depends on when we run the sky spot logs. If we run the `sky spot logs 2` before the `sky spot launch sleep 1000`, this logs will appear. : )

**Comment by Michaelvll** [comment]
Thank you for the observations @concretevitamin! I have addressed the comments, but I will tag this PR blocked until #1433 is merged, since that may change the logging logic.

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll.

**Comment by concretevitamin** [comment]
One minor thing:
```
» sky spot launch sleep 10000
...
I 11-27 03:16:03 log_lib.py:388] Start streaming logs for spot job 4.
INFO: Tip: use Ctrl-C to exit log streaming (task will not be killed).
INFO: Waiting for task resources on 1 node. This will block if the cluster is full.
INFO: All task resources reserved.
INFO: Reserved IPs: ['172.31.39.162']
```
Once the cluster is manually terminated on the console (simulating a preemption), the following lines immediately appear:
```
Shared connection to 3.134.104.105 closed.
⠦ Waiting for the job to start. It may take minutes.
```
Then the last line with the spinning cursor will disappear. It hangs for a few seconds, before showing:
```
ssh: connect to host 3.134.104.105 port 22: Connection timed out
⠹ Waiting for the job to start. It may take minutes.
```
the last line will somewhat quickly transition to
```
⠼ Waiting for the job to start (status: RECOVERING). It may take minutes.
```

The second snippet feels like new behavior? It's a bit weird to show it without either quickly transitioning to RECOVERING or showing "it may be preempted". Subjective, up to you.

**Comment by Michaelvll** [comment]
Thank you for the review @concretevitamin! I could not reproduce the issue you mentioned, but likely, it is because the time we wait for the spot status update is not long enough. I increased it to `2.1 * JOB_STATUS_CHECK_GAP_SECONDS`, could you test it again to see if the problem mentioned above still happens?

**Comment by concretevitamin** [comment]
I can still reproduce that snippet. Maybe we should debug live offline?

**Comment by Michaelvll** [comment]
Updated the message to the following when preempted. PTAL : )
image

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:300]: Maybe we can leave this part out? It's in the spinner line.

File: sky/spot/spot_utils.py
Comment (concretevitamin) [inline sky/spot/spot_utils.py:299]: I tried canceling a `sleep 1000` job and didn't see this message being printed. Saw it when manually terminating the spot VM.

Comment (Michaelvll) [reply sky/spot/spot_utils.py:299]: Ahh, good point. Since we recently changed the behavior of `sky spot cancel`, this will only happen for preemption.

File: sky/spot/spot_utils.py
Comment (Michaelvll) [reply sky/spot/spot_utils.py:300]: Good point! Removed it.

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll!

**Pull request merged**
